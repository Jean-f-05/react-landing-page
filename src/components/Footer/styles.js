import styled, { css } from 'styled-components';
import { Wrapper as FooterElement } from '../../components/TextComponent/styles';
import { Wrapper as SectionContainer } from '../../components/SectionContainer/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  text-align: center;
  border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration:none;
      font-size: ${theme.fonts.small};
      }

  & ${FooterElement} {
    font-size: ${theme.fonts.small};
}

  & ${SectionContainer}{
      padding-top: 0;
      padding-bottom: 0;
  }
`}`;
