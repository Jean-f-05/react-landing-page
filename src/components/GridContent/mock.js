export default {
  title: 'GRID CONTENT',
  html: `<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Zenonis est, inquam, hoc Stoici. Restinguet citius, si ardentem acceperit. Sed quanta sit alias, nunc tantum possitne esse tanta. Duo Reges: constructio interrete. </p>

  <h2>Similiter sensus, cum accessit ad naturam, tuetur illam quidem, sed etiam se tuetur;</h2>

  <p>Primum in nostrane potestate est, quid meminerimus? Quod iam a me expectare noli. Nihil ad rem! Ne sit sane; Sed haec omittamus; Praeteritis, inquit, gaudeo. </p>`,
  background: false,
};
