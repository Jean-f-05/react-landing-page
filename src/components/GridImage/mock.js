export default {
  background: false,
  title: 'My Grid',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipisicing elit.Ut quasi error, repellat quia, rerum placeat voluptate corrupti veritatis aliquam sapiente animi ipsa illum non vitae similique temporibus quod accusantium quam?',
  grid: [
    {
      altText: 'GRID 1',
      src: 'https://source.unsplash.com/random/300x300',
    },
    {
      altText: 'GRID 2',
      src: 'https://source.unsplash.com/random/300x300',
    },
    {
      altText: 'GRID 3',
      src: 'https://source.unsplash.com/random/300x300',
    },
    {
      altText: 'GRID 4',
      src: 'https://source.unsplash.com/random/300x300',
    },
    {
      altText: 'GRID 5',
      src: 'https://source.unsplash.com/random/300x300',
    },
    {
      altText: 'GRID 6',
      src: 'https://source.unsplash.com/random/300x300',
    },
  ],
};
