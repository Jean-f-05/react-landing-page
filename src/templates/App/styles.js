import styled, { css } from 'styled-components';

const changeBg = (color, theme) => css`
    background-color: ${color};
    color:${() => theme.colors.white};
`;

export const Wrapper = styled.div`
    ${({ color, theme }) => css`

       ${changeBg(color, theme)}
`}

    font-size: ${({ theme }) => theme.fonts.regular}
`;
