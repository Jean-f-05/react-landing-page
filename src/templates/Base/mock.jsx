import linksMock from '../../components/NavLinks/mock';
import GridText from '../../components/GridText';
import GridMock from '../../components/GridText/mock';

export const mockBase = {
  children: (
    <>
      <GridText {...GridMock} background />
      <GridText {...GridMock} />
      <GridText {...GridMock} background />
      <GridText {...GridMock} />
      <GridText {...GridMock} />
    </>
  ),
  links: linksMock,
  logoData: {
    text: 'Logo',
    link: '#',
  },
  footerHtml: '<p>Teste de footer</p>',
};
