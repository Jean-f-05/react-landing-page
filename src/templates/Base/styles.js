import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  padding-top: 5.4rem;

  @media ${theme.media.upToMedium} {
    padding-top: 0 ;
  }
`}`;
