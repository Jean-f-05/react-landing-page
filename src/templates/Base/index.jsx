import * as S from './styles';
import P from 'prop-types';
import Menu from '../../components/Menu';
import Footer from '../../components/Footer';
import GoTop from '../../components/GoTop';

const Base = ({ links, logoData, footerHtml, children }) => {
  return (
    <>
      <Menu links={links} logoData={logoData} />
      <S.Wrapper>
        {children}
        <Footer footerHtml={footerHtml} />
      </S.Wrapper>
      <GoTop />
    </>
  );
};

export default Base;

Base.propTypes = {
  children: P.node.isRequired,
  ...Menu.propTypes,
  footerHtml: P.string.isRequired,
};
