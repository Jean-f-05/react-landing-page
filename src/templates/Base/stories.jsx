import Base from '.';
import mock from './mock';
import { mockBase } from './mock';

export default {
  title: 'Templates/Base',
  component: Base,
  args: mockBase,
};

export const Template = (args) => {
  return (
    <div>
      <Base {...args}></Base>
    </div>
  );
};
