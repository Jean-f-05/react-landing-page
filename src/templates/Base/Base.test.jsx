import { renderTheme } from '../../styles/renderTheme';
import Base from '.';
import { screen } from '@testing-library/react';
import { mockBase } from './mock';

describe('<Base/>', () => {
  it('should ', () => {
    const { container } = renderTheme(<Base {...mockBase}></Base>);
    expect(container).toMatchSnapshot();
  });
});
