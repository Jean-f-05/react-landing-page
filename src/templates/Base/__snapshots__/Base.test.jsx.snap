// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base/> should  1`] = `
.c11 {
  padding-top: 5.4rem;
}

.c4 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c7 {
  color: #0A1128;
  font-size: 4rem;
  text-transform: uppercase;
}

.c14 {
  color: #fff;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c19 {
  color: #fff;
  font-size: 3.2rem;
  text-transform: none;
}

.c21 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c22 {
  color: #0A1128;
  font-size: 3.2rem;
  text-transform: none;
}

.c2 {
  position: fixed;
  z-index: 5;
  top: 0;
  right: 0;
  left: 0;
  border-bottom: #ddd;
  background-color: #fff;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c2 .c3 {
  padding-top: 0;
  padding-bottom: 0;
}

.c2 .c6 {
  margin-top: 0;
  margin-bottom: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  background-color: #fff;
  z-index: 10;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 4rem;
  height: 4rem;
  background-color: #0A1128;
  color: #fff;
  border: none;
  display: none;
  pointer-events: all;
}

.c0 svg {
  width: 2.5rem;
  height: 2.5rem;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c8 > img {
  max-height: 3rem;
}

.c9 {
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c10 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0A1128;
  position: relative;
}

.c10::after {
  content: "";
  position: absolute;
  bottom: 0.76rem;
  left: 50%;
  background-color: #dc143c;
  height: 0.2rem;
  width: 0;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c10:hover::after {
  left: 25%;
  width: 50%;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c16 {
  font-size: 2.4rem;
}

.c23 {
  text-align: center;
  border-top: 0.1rem solid #ddd;
}

.c23 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
}

.c23 .c15 {
  font-size: 1.6rem;
}

.c23 .c3 {
  padding-top: 0;
  padding-bottom: 0;
}

.c24 {
  position: fixed;
  background-color: #0A1128;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 10;
}

.c13 .c15 {
  margin-bottom: 5.6rem;
}

.c17 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(28rem,1fr));
}

.c17 gap:3.2rem @media (max-width:768px) {
  grid-template-columns: 1fr;
}

.c18 .c6 {
  position: relative;
  left: 5rem;
}

.c18 .c6:before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 7rem;
  top: -3rem;
  left: -5rem;
  -webkit-transform: rotate(5deg);
  -ms-transform: rotate(5deg);
  transform: rotate(5deg);
}

.c12 {
  background: #fff;
  color: #0A1128;
  background: #0A1128;
  color: #fff;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c20 {
  background: #fff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (max-width:768px) {
  .c11 {
    padding-top: 0;
  }
}

@media (max-width:768px) {
  .c14 {
    font-size: 4rem;
  }
}

@media (max-width:768px) {
  .c21 {
    font-size: 4rem;
  }
}

@media (max-width:768px) {
  .c2 {
    height: 100vh;
    visibility: hidden;
    opacity: 0;
  }

  .c2 .c3 {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    height: 100vh;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    margin: 0;
    overflow-y: auto;
  }

  .c2 .c6 {
    padding-bottom: 3.2rem;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c5 {
    display: block;
    text-align: center;
    padding: 4.8rem 0;
  }
}

@media (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:768px) {
  .c9 {
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6 c7"
        >
          <a
            class="c8"
            href="#"
          >
            Logo
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="c9"
        >
          <a
            class="c10"
            href="#target1"
            target="self"
          >
            Link 1
          </a>
          <a
            class="c10"
            href="#target2"
            target="self"
          >
            Link 2
          </a>
          <a
            class="c10"
            href="#target3"
            target="self"
          >
            Link 3
          </a>
          <a
            class="c10"
            href="#target4"
            target="self"
          >
            Link 4
          </a>
          <a
            class="c10"
            href="#target5"
            target="self"
          >
            Link 5
          </a>
          <a
            class="c10"
            href="#target6"
            target="self"
          >
            Link 6
          </a>
          <a
            class="c10"
            href="#target7"
            target="self"
          >
            Link 7
          </a>
          <a
            class="c10"
            href="#target8"
            target="self"
          >
            Link 8
          </a>
          <a
            class="c10"
            href="#target9"
            target="self"
          >
            Link 9
          </a>
          <a
            class="c10"
            href="#target10"
            target="self"
          >
            Link 10
          </a>
        </nav>
      </div>
    </div>
  </div>
  <div
    class="c11"
  >
    <div
      class="c12"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c13"
        >
          <h2
            class="c6 c14"
          >
            My Grid
          </h2>
          <div
            class="c15 c16"
          >
            Lorem ipsum dolor sit amet, consectetur adipisicing elit.Ut quasi error, repellat quia, rerum placeat voluptate corrupti veritatis aliquam sapiente animi ipsa illum non vitae similique temporibus quod accusantium quam?
          </div>
          <div
            class="c17"
          >
            <div
              class="c18"
            >
              <h3
                class="c6 c19"
              >
                GRID 1
              </h3>
              <div
                class="c15 c16"
              >
                Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus hic neque natus atque ducimus vero fugit. Accusamus sequi molestiae quisquam modi provident cum, ea, nostrum nihil maxime, sunt tempora explicabo.
              </div>
            </div>
            <div
              class="c18"
            >
              <h3
                class="c6 c19"
              >
                GRID 2
              </h3>
              <div
                class="c15 c16"
              >
                Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus hic neque natus atque ducimus vero fugit. Accusamus sequi molestiae quisquam modi provident cum, ea, nostrum nihil maxime, sunt tempora explicabo.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c20"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c13"
        >
          <h2
            class="c6 c21"
          >
            My Grid
          </h2>
          <div
            class="c15 c16"
          >
            Lorem ipsum dolor sit amet, consectetur adipisicing elit.Ut quasi error, repellat quia, rerum placeat voluptate corrupti veritatis aliquam sapiente animi ipsa illum non vitae similique temporibus quod accusantium quam?
          </div>
          <div
            class="c17"
          >
            <div
              class="c18"
            >
              <h3
                class="c6 c22"
              >
                GRID 1
              </h3>
              <div
                class="c15 c16"
              >
                Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus hic neque natus atque ducimus vero fugit. Accusamus sequi molestiae quisquam modi provident cum, ea, nostrum nihil maxime, sunt tempora explicabo.
              </div>
            </div>
            <div
              class="c18"
            >
              <h3
                class="c6 c22"
              >
                GRID 2
              </h3>
              <div
                class="c15 c16"
              >
                Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus hic neque natus atque ducimus vero fugit. Accusamus sequi molestiae quisquam modi provident cum, ea, nostrum nihil maxime, sunt tempora explicabo.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c12"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c13"
        >
          <h2
            class="c6 c14"
          >
            My Grid
          </h2>
          <div
            class="c15 c16"
          >
            Lorem ipsum dolor sit amet, consectetur adipisicing elit.Ut quasi error, repellat quia, rerum placeat voluptate corrupti veritatis aliquam sapiente animi ipsa illum non vitae similique temporibus quod accusantium quam?
          </div>
          <div
            class="c17"
          >
            <div
              class="c18"
            >
              <h3
                class="c6 c19"
              >
                GRID 1
              </h3>
              <div
                class="c15 c16"
              >
                Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus hic neque natus atque ducimus vero fugit. Accusamus sequi molestiae quisquam modi provident cum, ea, nostrum nihil maxime, sunt tempora explicabo.
              </div>
            </div>
            <div
              class="c18"
            >
              <h3
                class="c6 c19"
              >
                GRID 2
              </h3>
              <div
                class="c15 c16"
              >
                Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus hic neque natus atque ducimus vero fugit. Accusamus sequi molestiae quisquam modi provident cum, ea, nostrum nihil maxime, sunt tempora explicabo.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c20"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c13"
        >
          <h2
            class="c6 c21"
          >
            My Grid
          </h2>
          <div
            class="c15 c16"
          >
            Lorem ipsum dolor sit amet, consectetur adipisicing elit.Ut quasi error, repellat quia, rerum placeat voluptate corrupti veritatis aliquam sapiente animi ipsa illum non vitae similique temporibus quod accusantium quam?
          </div>
          <div
            class="c17"
          >
            <div
              class="c18"
            >
              <h3
                class="c6 c22"
              >
                GRID 1
              </h3>
              <div
                class="c15 c16"
              >
                Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus hic neque natus atque ducimus vero fugit. Accusamus sequi molestiae quisquam modi provident cum, ea, nostrum nihil maxime, sunt tempora explicabo.
              </div>
            </div>
            <div
              class="c18"
            >
              <h3
                class="c6 c22"
              >
                GRID 2
              </h3>
              <div
                class="c15 c16"
              >
                Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus hic neque natus atque ducimus vero fugit. Accusamus sequi molestiae quisquam modi provident cum, ea, nostrum nihil maxime, sunt tempora explicabo.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c20"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c13"
        >
          <h2
            class="c6 c21"
          >
            My Grid
          </h2>
          <div
            class="c15 c16"
          >
            Lorem ipsum dolor sit amet, consectetur adipisicing elit.Ut quasi error, repellat quia, rerum placeat voluptate corrupti veritatis aliquam sapiente animi ipsa illum non vitae similique temporibus quod accusantium quam?
          </div>
          <div
            class="c17"
          >
            <div
              class="c18"
            >
              <h3
                class="c6 c22"
              >
                GRID 1
              </h3>
              <div
                class="c15 c16"
              >
                Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus hic neque natus atque ducimus vero fugit. Accusamus sequi molestiae quisquam modi provident cum, ea, nostrum nihil maxime, sunt tempora explicabo.
              </div>
            </div>
            <div
              class="c18"
            >
              <h3
                class="c6 c22"
              >
                GRID 2
              </h3>
              <div
                class="c15 c16"
              >
                Lorem, ipsum dolor sit amet consectetur adipisicing elit. Temporibus hic neque natus atque ducimus vero fugit. Accusamus sequi molestiae quisquam modi provident cum, ea, nostrum nihil maxime, sunt tempora explicabo.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c23"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c15 c16"
        >
          <p>
            Teste de footer
          </p>
        </div>
      </div>
    </div>
  </div>
  <a
    aria-label="Go to top"
    class="c24"
    href="#"
    title="Got to top"
  >
    <svg
      aria-hidden="true"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
      />
    </svg>
  </a>
</div>
`;
